actnow/accounts/serializers.py

```diff
@@ -13,7 +13,32 @@ class Meta:
     def create(self, validated_data):
         if not validated_data["username"]:
             del validated_data["username"]
-        user = User.objects.create_user(**validated_data)
+
+        request_data = self.context["request"].data
+
+        first_name = request_data.get("first_name", validated_data["username"])
+        last_name = request_data.get("last_name", "")
+        bio = request_data.get("bio", "")
+        photo = request_data.get("photo", "")
+        location = request_data.get("location", "")
+        phone_number = request_data.get("phone_number", "")
+        social_media_link = request_data.get("social_media_link", "")
```

Comment:
accounts shouldn't know anything about these fields... it should just pass validated_data as before.

Reply:
The reason I'm doing this is because
Validated data will only contain User data i.e username, password and email as specified in the User serializer. Unless we modify the user serializer(which is still in accounts) to have user profile data, this is the only way to pass user profile data to the instance

Reply:
@kilemensi  I've tested with passing just `request_data` and the issue is how the data looks like. e.g here is how the data looks like.
```
{'username': ['esirne'], 'password': ['adgjshahdhgjsda$4234assa1d'], 'email': ['nyambs@esir.cozxa'], 'first_name': ['Blalalal'], 'last_name': ['Blalalal']}
```
Without using a Serializer, fields will be saved as they are i.e including the `[ ]`.

Reply:
Just thinking out loud; This seems fine to me... Whoever needs this data down the line (in this case **profiles** app), will use their own `Serializer` to deal with it.

But @esirK , the key design goal is to separate these apps i.e. accounts shouldn't know anything about profiles, profiles shouldn't know anything about another app that uses it down the line, etc.

So think about it and if there are other ways, lets discuss them now. One way I can think of will require changing our APIs: allowing apps to create profiles for users but if it's app creating the profile, it must pass the user to whom the profile belongs to as part of parameters. The good thing about this approach is every call is self contained but the bad thing is complexity of API permissions as well as the client will have to handle making multiple API calls.

The good thing about this PR's approach is it's one API call and everything gets initialized for users seamlessly. From this point onwards, we no longer need app account to do anything for users. Users will use their own accounts to do the rest.

So out of these 2 approaches, I kind of prefer this PR's approach.

Reply:
@kilemensi  your comment makes perfect sense. I've updated the code. You can re-review it as I continue looking at things I might have missed.